contracts/svm/testing/tests/swap.rs

```diff
@@ -0,0 +1,568 @@
+            svm,
+            &instructions,
+            &self.mint_authority,
+            &[&self.mint_authority],
+        )
+        .unwrap();
+    }
+
+    pub fn token_balance_matches(svm: &mut LiteSVM, account: &Pubkey, amount: u64) -> bool {
+        let token_account_option = &mut svm.get_account(account).map(|account| {
+            anchor_spl::token_interface::TokenAccount::try_deserialize(&mut account.data.as_slice())
+                .unwrap()
+        });
+
+        if token_account_option.is_none() {
+            return amount == 0;
+        }
+
+        token_account_option.unwrap().amount == amount
+    }
+
+    pub fn create_token_account(&self, svm: &mut LiteSVM, owner: &Pubkey) -> Pubkey {
+        let token_account = Keypair::new();
+        let token_account_rent =
+            svm.minimum_balance_for_rent_exemption(spl_token_2022::state::Account::LEN);
+        let instructions = vec![
+            system_instruction::create_account(
+                &self.mint_authority.pubkey(),
+                &token_account.pubkey(),
+                token_account_rent,
+                spl_token_2022::state::Account::LEN.try_into().unwrap(),
+                &self.token_program,
+            ),
+            initialize_account2(
+                &self.token_program,
+                &token_account.pubkey(),
+                &self.mint,
+                owner,
+            )
+            .unwrap(),
+        ];
+        submit_transaction(
+            svm,
+            &instructions,
+            &self.mint_authority,
+            &[&token_account, &self.mint_authority],
+        )
+        .unwrap();
+        token_account.pubkey()
+    }
+
+    pub fn new(svm: &mut LiteSVM, token_program: Pubkey, decimals: u8) -> Self {
+        let mint = Keypair::new();
+        let mint_authority = generate_and_fund_key(svm);
+        let mint_rent = svm.minimum_balance_for_rent_exemption(spl_token_2022::state::Mint::LEN);
+        let instructions = vec![
+            system_instruction::create_account(
+                &mint_authority.pubkey(),
+                &mint.pubkey(),
+                mint_rent,
+                spl_token_2022::state::Mint::LEN.try_into().unwrap(),
+                &token_program,
+            ),
+            spl_token_2022::instruction::initialize_mint(
+                &token_program,
+                &mint.pubkey(),
+                &mint_authority.pubkey(),
+                None,
+                decimals,
+            )
+            .unwrap(),
+        ];
+        submit_transaction(
+            svm,
+            &instructions,
+            &mint_authority,
+            &[&mint, &mint_authority],
+        )
+        .unwrap();
+        Self {
+            mint: mint.pubkey(),
+            decimals,
+            mint_authority,
+            token_program,
+        }
+    }
+
+    pub fn get_amount_with_decimals(&self, amount: f64) -> u64 {
+        (amount * 10_f64.powi(self.decimals as i32)).floor() as u64
```

Comment:
why not do all of this in u64?

Reply:
also is this not `10f64`?

Reply:
I run my tests with different values for the number of decimals, so It's useful to have a number-of-decimal-agnostic representation. 
Also in my asserts I need decimal numbers, thus the float.